src/query.js

```diff
@@ -31,6 +32,70 @@ function toQueryString(obj) {
   return qs;
 }
 
+const findStylePreset = (id, stylePresets) => {
+  const flatPresets = [
+    ...stylePresets.filter(p => p.type !== 'sublist'),
+    ...stylePresets.filter(p => p.type === 'sublist').flatMap(p => p.presets),
+  ];
+
+  return flatPresets.find(preset => preset.id === id);
+};
+
+const findBranchPattern = (map, branchPatterns) => {
+  const branchPattern = branchPatterns.find(p => p.id === map.branchPatternId);
```

Comment:
I'm not totally sure I follow why we need `branchPatternId` when it's the same as the `id` field here

Reply:
`id` gets overloaded in the `MapStyleInputWrapper` in a way that might get confusing (`id` of the branch pattern itself or the style...), but I hear you--I changed it to use `branchId` instead since that already existed in `MapStyleInputWrapper`.